bin/main.ml

```diff
@@ -769,10 +843,67 @@ let build_targets =
     in
     set_common common ~targets;
     let log = Log.create common in
-    Scheduler.go ~log ~common
-      (Main.setup ~log common >>= fun setup ->
+
```

Comment:
The polling mode is useful for both `dune build` and `dune runtest`. Could you factorize the code so that it works with both commands?